zerocopy-derive/src/lib.rs

```diff
@@ -62,10 +73,112 @@ use {crate::ext::*, crate::repr::*};
 #[proc_macro_derive(KnownLayout)]
 pub fn derive_known_layout(ts: proc_macro::TokenStream) -> proc_macro::TokenStream {
     let ast = syn::parse_macro_input!(ts as DeriveInput);
+
+    let is_repr_c_struct = match &ast.data {
+        Data::Struct(..) => {
+            let reprs = try_or_print!(repr::reprs::<Repr>(&ast.attrs));
+            if reprs.iter().any(|(_meta, repr)| repr == &Repr::C) {
+                Some(reprs)
+            } else {
+                None
+            }
+        }
+        Data::Enum(..) | Data::Union(..) => None,
+    };
+
+    let fields = ast.data.field_types();
+    let fields = fields.split_last();
```

Comment:
Maybe a more descriptive name for this variable or inline this into its usage site below?